src/platform/qt/src/config.hpp

```diff
@@ -125,7 +136,13 @@ struct QtConfig final : nba::PlatformConfig {
       }
     #endif
 
-    return "config.toml";
+    #ifdef PORTABLE_MODE
+      return "config.toml";
+    #else
+      auto config_directory = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation).toStdString();
+      const std::filesystem::path config_path = config_directory;
+      return  (config_path / "NanoBoyAdvance" / "config.toml").string();
+    #endif

```

Comment:
Would it make sense to use the new code path using QStandardPaths also for macOS app bundles? If so the method could be shrinked down a bit then and the condition could be `#if defined(PORTABLE_MODE) && !defined(MACOS_BUILD_APP_BUNDLE)`